fcm/fcm.go

```diff
@@ -90,31 +90,44 @@ func (f *FCM) ConnectToFcm(fcmNotifications chan<- notification.PrinterNotificat
 	resp, err := http.Get(fmt.Sprintf("%s?token=%s", f.fcmServerBindURL, iidToken))
 	if err != nil {
 		// failed for ever no need to retry
+		log.Errorf("%v", err)
 		quit <- struct{}{}
 		return err
 	}
 	if resp.StatusCode == 200 {
 		reader := bufio.NewReader(resp.Body)
 		go func() {
 			for {
-				sizeB, err := reader.ReadBytes('\n')
-				if err == io.EOF {
-					log.Info("DRAIN message received, client reconnecting.")
-					dead <- struct{}{}
-					break
+				raw_input, err1 := reader.ReadBytes('\n')
+				inputA := strings.SplitN(string(raw_input), "\\n", 2)
+				notification_size := inputA[0]
+				notification_data1 := ""
+				if len(inputA) > 1 {
```

Comment:
should this be > 0?

Reply:
we are trying to access second element of array so size should be > 1